backend/sources/chat-plugin/src/main/java/co/airy/core/chat_plugin/Stores.java

```diff
@@ -42,9 +45,23 @@
     private void startStream() {
         final StreamsBuilder builder = new StreamsBuilder();
 
-        builder.<String, Message>stream(applicationCommunicationMessages)
+        final KStream<String, Message> messageStream = builder.<String, Message>stream(applicationCommunicationMessages)
+                .filter((messageId, message) -> "chat_plugin".equals(message.getSource()));
+
+        // Echoes
+        messageStream.filter((messageId, message) -> message.getSenderType().equals(SenderType.SOURCE_CONTACT))
                 .peek((messageId, message) -> webSocketController.onNewMessage(message));
 
+        messageStream.filter((messageId, message) -> !message.getSenderType().equals(SenderType.SOURCE_CONTACT)
+                && message.getDeliveryState().equals(DeliveryState.PENDING)
+        )
+                .mapValues((messageId, message) -> {
+                    message.setDeliveryState(DeliveryState.DELIVERED);
+                    webSocketController.onNewMessage(message);
+                    return message;
+                })
```

Comment:
if something goes wrong with the websocket, we end up not changing the state of the message so I'm thinking we should do the websocket part in a terminal node